subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -1186,17 +1228,28 @@ static int bt_bap_scan_delegator_init(void)
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 2 */
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 1 */
 
-	bt_le_per_adv_sync_cb_register(&pa_sync_cb);
+	if (!pa_sync_cb_registered) {
+		bt_le_per_adv_sync_cb_register(&pa_sync_cb);
+		pa_sync_cb_registered = true;
+	}
 
-	return 0;
-}
+	scan_delegator_cbs = cb;
+	scan_delegator_registered = true;
 
-SYS_INIT(bt_bap_scan_delegator_init, APPLICATION, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
+	return bt_bap_bass_register();
+}
 
-/****************************** PUBLIC API ******************************/
-void bt_bap_scan_delegator_register_cb(struct bt_bap_scan_delegator_cb *cb)
+int bt_bap_scan_delegator_unregister(void)
 {
-	scan_delegator_cbs = cb;
+	if (!scan_delegator_registered) {
+		LOG_DBG("Scan delegator not yet registered\n");
+		return -EALREADY;
+	}
+
+	scan_delegator_cbs = NULL;
+	scan_delegator_registered = false;
+
+	return bt_bap_bass_unregister();
```

Comment:
If `bt_bap_bass_unregister` fails we probably don't want to reset `scan_delegator_cbs` and `scan_delegator_registered`

Reply:
That sounds true, corrected.